core/src/types/tokenizer.rs

```diff
@@ -0,0 +1,36 @@
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum TiktokenTokenizerBase {
+    #[serde(rename = "o200k_base")]
+    O200kBase,
+    #[serde(rename = "cl100k_base")]
+    Cl100kBase,
+    #[serde(rename = "p50k_base")]
+    P50kBase,
+    #[serde(rename = "r50k_base")]
+    R50kBase,
+    #[serde(rename = "anthropic_base")]
+    AnthropicBase,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub enum SentencePieceTokenizerBase {
+    #[serde(rename = "model_v1")]
```

Comment:
I don't think these renames are useful, the rename_all is doing exactly this I think